packages/contracts/token_contract/src/main.sw

```diff
@@ -0,0 +1,167 @@
+contract;
+
+dep errors;
+dep events;
+dep interface;
+
+use string::String;
+use std::{
+    auth::{
+        msg_sender,
+    },
+    call_frames::contract_id,
+    constants::{
+        BASE_ASSET_ID,
+        ZERO_B256,
+    },
+    context::{
+        balance_of,
+        msg_amount,
+    },
+    token::{
+        mint_to_contract,
+        transfer,
+    },
+};
+
+use errors::{DepositError, InitError, MintError};
+use events::{DepositEvent, InitialMintEvent, MintEvent, TransferBackEvent};
+use interface::TokenContract;
+
+storage {
+    // Contract owner identity
```

Comment:
Using `//` is for regular comments while `///` is for documentation comments. I suggest using doc comments instead of regular comments